src/wallet/walletdb.cpp

```diff
@@ -803,6 +803,12 @@ static DBErrors LoadDescriptorWalletRecords(CWallet* pwallet, DatabaseBatch& bat
         }
         pwallet->LoadDescriptorScriptPubKeyMan(id, desc);
 
+        // Prior to doing anything with this spkm, verify ID compatibility
+        if (id != pwallet->GetDescriptorScriptPubKeyMan(desc)->GetID()) {
+            strErr = "The descriptor ID calculated by the wallet differs from the one in DB";
+            return DBErrors::CORRUPT;
```

Comment:
Why not `UNKNOWN_DESCRIPTOR` instead?

Reply:
The descriptor isn't unknown. It's a corruption error as the id written in the database doesn't match the id that we calculate.